src/web-components/who-is-online/components/dropdown.ts

```diff
@@ -0,0 +1,174 @@
+    const dropdownCta = slotDropdown.assignedElements()[0];
+    const hasDropdownContent = elements.includes(dropdownContent);
+    const hasDropdownList = elements.includes(dropdownList);
+    const hasDropdownCta = elements.includes(dropdownCta);
+
+    if (!(hasDropdownContent || hasDropdownList || hasDropdownCta)) {
+      this.open = false;
+      this.selected = '';
+      this.emitEvent('clickout', {
+        detail: {
+          open: this.open,
+        },
+        bubbles: false,
+        composed: false,
+      });
+    }
+  };
+
+  private close = () => {
+    this.emitEvent('close', {
+      bubbles: false,
+      composed: false,
+    });
+  };
+
+  public dropdownOptionsHandler = ({ detail }: CustomEvent) => {};
+
+  private selectParticipant = (participantId: string) => {
+    return () => {
+      this.selected = participantId;
+    };
+  };
+
+  private renderParticipants() {
+    if (!this.participants) return;
+
+    const options = Object.values(WhoIsOnlineDropdownOptions).map((label) => {
+      return { label };
+    });
+
+    const icons = ['place', 'send'];
+
+    return this.participants.map((participant) => {
+      const letterColor = this.textColorValues.includes(participant.slotIndex)
+        ? '#FFFFFF'
+        : '#000000';
+
+      const contentClasses = {
+        'who-is-online-dropdown__content': true,
+        'who-is-online-dropdown__content--selected': this.selected === participant.id,
+      };
+
+      return html`
+        <superviz-dropdown
+        options=${JSON.stringify(options)}
+        label="label"
+        returnTo="label"
+        position="bottom-right"
+        @selected=${this.dropdownOptionsHandler}
+        icons="${JSON.stringify(icons)}"
+        >
+        <div class=${classMap(contentClasses)} @click=${this.selectParticipant(
+        participant.id,
+      )} slot="dropdown">
+          <div class="who-is-online-dropdown__participant" style="border-color: 
+          ${participant.color}">
+              <div class="who-is-online-dropdown__avatar" style="background-color: ${
+                participant.color
+              }; color: ${letterColor}">
+                ${participant.name?.at(0)}
+              </div>
+            </div>
+            <span class="user-name">${participant.name}</span>
+            <superviz-icon class="icon" name="right" color="var(--sv-gray-600)"></superviz-icon>
+          </div>
+        </div>
+      </superviz-dropdown>
+      `;
+    });
+  }
+
+  protected render() {
+    const menuClasses = {
+      menu: true,
+      'menu--bottom-left': this.position === 'bottom-left',
+      'menu--bottom-center': this.position === 'bottom-center',
+      'menu--bottom-right': this.position === 'bottom-right',
+      'menu-open': this.open,
+      'menu-left': this.align === 'left',
+      'menu-right': this.align === 'right',
+    };
+
+    const toggle = () => {
+      this.open = !this.open;
+      this.selected = '';
+    };
```

Comment:
If you put the function inside the render, you re-render this function every time the component is updated. You can use getters on 'menuClasses' and a private method to toggle the